### PR TITLE
feat(manifest-service): Add GitService with GetCommitInfo RPC to manifest-export-service

### PR DESCRIPTION
**Pull request by AminSlk**
Ref: SRX-4DRR81

### CODE REVIEW

File: services/manifest-repo-export-service/pkg/service/git_test.go [hunk 1]
```diff
@@ -0,0 +1,1140 @@
...
+			}
+			ctx := rp.AddGeneratorToContext(testutil.MakeTestContext(), uuidGenerate)
+			dbHandler := repo.State().DBHandler
+
+			err := dbHandler.WithTransactionR(ctx, 0, false, func(ctx context.Context, transaction *sql.Tx) error {
+				err := dbHandler.DBWriteMigrationsTransformer(ctx, transaction)
+				if err != nil {
+					return err
+				}
+				err = dbHandler.DBInsertApplication(ctx, transaction, "app", 1, db.AppStateChangeCreate, db.DBAppMetaData{Team: "team"})
```
Comment (sven-urbanski-freiheit-com) [inline services/manifest-repo-export-service/pkg/service/git_test.go:946]: this test is very big. Is there a way to split it up?

File: services/manifest-repo-export-service/pkg/service/git_test.go [hunk 2]
```diff
@@ -0,0 +1,1140 @@
...
+					return err
+				}
+				err = dbHandler.DBInsertApplication(ctx, transaction, "app-2", 2, db.AppStateChangeCreate, db.DBAppMetaData{Team: "team"})
+				if err != nil {
+					return err
+				}
+				err = dbHandler.DBInsertApplication(ctx, transaction, "app-3", 3, db.AppStateChangeCreate, db.DBAppMetaData{Team: "team"})
+				if err != nil {
+					return err
+				}
```
Comment (sven-urbanski-freiheit-com) [inline services/manifest-repo-export-service/pkg/service/git_test.go:961]: this could be a for-loop

File: services/manifest-repo-export-service/pkg/service/git_test.go [hunk 3]
```diff
@@ -0,0 +1,1140 @@
...
+					},
+				})
+				if err != nil {
+					return err
+				}
+				err = dbHandler.DBWriteEnvironment(ctx, transaction, "development-3", config.EnvironmentConfig{
+					Upstream: &config.EnvironmentConfigUpstream{
+						Latest: true,
+					},
+				})
```
Comment (sven-urbanski-freiheit-com) [inline services/manifest-repo-export-service/pkg/service/git_test.go:1068]: why are these necessary? We already have the `environmentSetup`, is that not sufficient?

File: services/manifest-repo-export-service/pkg/service/git_test.go (hunk 1 above)
Comment (AminSlk) [reply services/manifest-repo-export-service/pkg/service/git_test.go]: Moved the DB setup parts to a new function now.
The rest is the same as the cd service

File: services/manifest-repo-export-service/pkg/service/git_test.go (hunk 2 above)
Comment (AminSlk) [reply services/manifest-repo-export-service/pkg/service/git_test.go]: Fixed

File: services/manifest-repo-export-service/pkg/service/git_test.go (hunk 3 above)
Comment (AminSlk) [reply services/manifest-repo-export-service/pkg/service/git_test.go]: The `environmentSetup` only consists of transformers. In order for the transformers to work in the `manifest-repo-export-service`, we need to have some rows in the database(which is actually the `cd-service`'s job).
We had the same approach here:
https://github.com/freiheit-com/kuberpult/blob/231e17bfb039ff752df784f1a8e5a4a4b0414bdb/services/manifest-repo-export-service/pkg/repository/transformer_test.go#L369

**Review by sven-urbanski-freiheit-com — APPROVED** [review]
